File: src/main/webapp/static/JavaScripts/CodeJsonizer.js
function fillJsonTextArea(){
    var jsontextarea = document.getElementById("jsonTextarea");
    var code=document.getElementById("Code").value;
    var description=document.getElementById("Describe").value;
    jsontextarea.value=codeToJson(code,description);
}

function fillCodeAndDescription(){
    var descriptionInput=document.getElementById("Describe");
    var codeInput=document.getElementById("Code");
    var json= document.getElementById("jsonTextarea").value;
    descriptionInput.value=jsonToDescription(json);
    codeInput.value=jsonToCode(json);
}

function codeToJson(code, description) {
    if (code.length !== 36) return "Wrong code format!";
    if (description==null)description=" ";

    var codeJson = {
        "GeographyCode": code.substring(0, 12),
        "TimeCode": code.substring(12, 26),
        "SourceCode": code.substring(26, 29),
        "CarrierCode": code.substring(29, 30),
        "DisasterCode": code.substring(30, 36),
        "Description": description
    };

    return JSON.stringify(codeJson,null,2);
}

function jsonToCode(json) {
    if (!ifJson(json)) return "Json format error.";

    var codeJson = JSON.parse(json);
    return codeJson.GeographyCode + codeJson.TimeCode + codeJson.SourceCode + codeJson.CarrierCode + codeJson.DisasterCode;
}

function jsonToDescription(json) {
    if (!ifJson(json)) return "Json format error.";

    var codeJson = JSON.parse(json);
    return codeJson.Description;
}

function ifJson(input) {
    try {
        JSON.parse(input);
        return true;
    } catch (error) {
        return false;
    }
}

File: src/main/java/com/qiu/Rook/CodeJsonizer.java
package com.qiu.Rook;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import com.google.gson.JsonSyntaxException;
import lombok.var;

public class CodeJsonizer {
    static public String codeToJson(String code,String description){
        if(code.length()!=36)return "wrong code format";
        JsonObject codeJson=new JsonObject();
        codeJson.addProperty("GeographyCode",code.substring(0,12));
        codeJson.addProperty("TimeCode",code.substring(12,26));
        codeJson.addProperty("SourceCode",code.substring(26,29));
        codeJson.addProperty("CarrierCode",code.substring(29,30));
        codeJson.addProperty("DisasterCode",code.substring(30,36));
        codeJson.addProperty("Description",description);
        return codeJson.toString();
    }

    static public String jsonToCode(String json){
        if(!ifJson(json))return "Json format error.";
        JsonObject codeJson= JsonParser.parseString(json).getAsJsonObject();
        return codeJson.get("GeographyCode").toString()+codeJson.get("TimeCode").toString()+codeJson.get("SourceCode").toString()+codeJson.get("CarrierCode").toString()+codeJson.get("DisasterCode").toString();
    }

    static public String jsonToDescription(String json){
        if(!ifJson(json))return "Json format error.";
        JsonObject codeJson= JsonParser.parseString(json).getAsJsonObject();
        return codeJson.get("Description").toString();
    }

    static private Boolean ifJson(String input){
        try {
            JsonParser parser = new JsonParser();
            JsonParser.parseString(input);
            return true;
        } catch (JsonSyntaxException e) {
            return false;
        }
    }
}
